lib/common/http1client.c

```diff
@@ -705,6 +708,7 @@ static void on_connection_ready(struct st_h2o_http1client_t *client)
         if (client->_is_chunked) {
             assert(body.base != NULL);
             size_t bytes;
+            assert(PTLS_ELEMENTSOF(reqbufs) - reqbufcnt >= 3); /* encode_chunk could write to 3 additional elements */
```

Comment:
FWIW, we could move this assert to immediately before the call to `h2o_socket_write`, that checks that `reqbufcnt` is no greater than `PTLS_ELEMENTSOF(reqbufs)`, under the assumption that buffer overrun on local variables do not cause harm until you exit from the function.

Not that we have to change the approach, but that's one way of checking the condition more broadly.

Reply:
👍 I thought of a similar approach, but was not 100% sure if it was okay to let overrun happen then check.